src/soilbiogeochem/SoilBiogeochemVerticalProfileMod.F90

```diff
@@ -168,45 +167,66 @@ subroutine SoilBiogeochemVerticalProfile(bounds, num_soilc,filter_soilc,num_soil
          endif
-
       end do
 
       !! aggregate root profile to column
       ! call p2c (decomp, nlevdecomp_full, &
       !      cinput_rootfr(bounds%begp:bounds%endp, :), &
       !      col_cinput_rootfr(bounds%begc:bounds%endc, :), &
       !      'unity')
-      do pi = 1,maxsoil_patches
-         do fc = 1,num_soilc
-            c = filter_soilc(fc)
-            if (pi <=  col%npatches(c)) then
-               p = col%patchi(c) + pi - 1
-               do j = 1,nlevdecomp
-                  col_cinput_rootfr(c,j) = col_cinput_rootfr(c,j) + cinput_rootfr(p,j) * patch%wtcol(p)
-               end do
-            end if
+      if(num_soilp>0)then
+         do pi = 1,maxsoil_patches
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               if(.not.col%is_fates(c))then
+                  if (pi <=  col%npatches(c)) then
+                     p = col%patchi(c) + pi - 1
+                     do j = 1,nlevdecomp
+                        col_cinput_rootfr(c,j) = col_cinput_rootfr(c,j) + cinput_rootfr(p,j) * patch%wtcol(p)
+                     end do
+                  end if
+               end if
+            end do
          end do
-      end do
+      end if
 
       ! repeat for column-native profiles: Ndep and Nfix
       do fc = 1,num_soilc
          c = filter_soilc(fc)
          rootfr_tot = 0._r8
          surface_prof_tot = 0._r8
-         ! redo column ntegration over active layer for column-native profiles
-         do j = 1, min(max(altmax_lastyear_indx(c), 1), nlevdecomp)
-            rootfr_tot = rootfr_tot + col_cinput_rootfr(c,j) * dzsoi_decomp(j)
-            surface_prof_tot = surface_prof_tot + surface_prof(j) * dzsoi_decomp(j)
-         end do
-         if ( (altmax_lastyear_indx(c) > 0) .and. (rootfr_tot > 0._r8) .and. (surface_prof_tot > 0._r8) ) then
-            do j = 1,  min(max(altmax_lastyear_indx(c), 1), nlevdecomp)
-               nfixation_prof(c,j) = col_cinput_rootfr(c,j) / rootfr_tot
-               ndep_prof(c,j) = surface_prof(j)/ surface_prof_tot
+         if(col%is_fates(c))then
+            ! For FATES, we just use the e-folding depth for both fixation and deposition
+            ! partially because the fixation may be free-living depending on FATES-side
+            ! fixation choices, and partially for simplicity
+            do j = 1, min(max(altmax_lastyear_indx(c), 1), nlevdecomp)
+               surface_prof_tot = surface_prof_tot + surface_prof(j) * dzsoi_decomp(j)
             end do
+            if ( (altmax_lastyear_indx(c) > 0) .and. (surface_prof_tot > 0._r8) ) then
+               do j = 1,  min(max(altmax_lastyear_indx(c), 1), nlevdecomp)
+                  nfixation_prof(c,j) = surface_prof(j)/ surface_prof_tot
+                  ndep_prof(c,j) = surface_prof(j)/ surface_prof_tot
+               end do
+            else
+               nfixation_prof(c,1) = 1./dzsoi_decomp(1)
+               ndep_prof(c,1) = 1./dzsoi_decomp(1)
+            endif
          else
```

Comment:
maybe add an end_if_fates comment as we have in other points of the code

Reply:
will do

Reply:
Done